### PR TITLE
fix(seo): inject per-page canonical + og meta on static-routed pages

### PR DESCRIPTION
**Pull request by BreakableHoodie**
## Summary

Google Search Console flagged **"Duplicate, Google chose different canonical than user"**. Root-caused to a conflict between two meta layers, and fixed by extending the SSR meta injection this repo already had.

`frontend/index.html` hardcodes the homepage's identity meta. `react-helmet-async` only manages tags it created itself, so client-side it **appends** a page-specific copy instead of replacing the default. Measured live on `/artists`:

| Tag | Count | Values (DOM order) |
|---|---|---|
| `link[rel=canonical]` | 1 | `/artists` ✅ |
| `og:url` | **2** | `https://settimes.ca/` **first**, then `/artists` ❌ |
| `og:title` | **2** | generic first ❌ |
| `description` | **2** | generic first ❌ |

Google reads `og:url` as a canonicalization hint and prefers the first tag, so every static page declared its own canonical while simultaneously claiming to be the homepage. All nine also served a **byte-identical 4217-byte shell**, reinforcing the duplicate cluster.

**Duplication is harmless when values AGREE and fatal when they disagree.** `/band/199` renders `canonical` twice — both correct — and Google is fine with it. The SSR routes already avoided the conflict because `serveWithInjectedMeta()` strips the defaults via `DEFAULT_META_RE` before injecting. That fix was only ever wired to `/event/*`, `/band/*`, `/venue/*`.

## Changes

- `functions/utils/staticPageMeta.js` — `STATIC_PAGES` registry + `staticPageHandler()` factory; 8 two-line route files. Values copied **verbatim** from each page's own `<Helmet>` so server and client agree.
- `/` deliberately excluded — `index.html`'s defaults *are* the homepage's correct meta.
- `_routes.json` — listed `/privacy`, `/terms`, `/about`, `/contact`, `/stats` in **both** `include` and `exclude`. Exclude wins, so those functions would never have been dispatched.
- All URLs built from `CANONICAL_HOST`, never `request.url` — preview deploys must not self-canonicalise.

## Sweep for siblings

- **`EventRecapPage.jsx` had no canonical at all** — the only page missing one. Recap URLs enter the sitemap the day after an event.
- A client-side canonical alone would have left recap pages with **the exact bug this PR fixes**, so `/events/:slug/recap` also gets a D1-backed Pages Function (modeled on `event/[slug].js`, same publish gating and `env.ASSETS.fetch` fallback discipline).

## Durable guard

`functions/__tests__/staticPageMeta.test.js` scans `_routes.json`, dynamically imports each route file, invokes `onRequestGet`, and asserts the emitted canonical **and** `og:url` equal `CANONICAL_HOST + path` — so a handler wired to the wrong path fails, not just a missing file. Mutation-proved: pointing `artists.js` at `/venues` fails with `expected 'https://settimes.ca/venues' to be 'https://settimes.ca/artists'`.

The shell fixture is `readFileSync(frontend/index.html)` rather than a copy, so a new default meta tag can't drift past the test. All 11 identity tags are asserted exactly-once (previously 3 of 11).

## Test plan

- [x] Backend: `lint` 0 errors · **102 files, 1041 tests pass**
- [x] Frontend: `lint` / `format:check` 0 errors · **85 files, 968 tests pass** · `build` ✓
- [x] `make review` (CodeRabbit) run pre-open; all 3 findings fixed in-branch
- [x] **Real `wrangler pages dev`** — mocked `ASSETS.fetch` proves the handler *builds* HTML, not that Cloudflare *dispatches* to it. All 8 paths return exactly 1 canonical / 1 og:url / 1 description with correct values; homepage default stripped; no `.pages.dev` leak; `/` unaffected
- [x] Recap route verified happy-path against live local D1, plus unknown-slug and gate-closed fallbacks

## Notes / follow-ups

- Recap SSR gates on `(is_published = 1 OR status = 'archived')` to match `event/[slug].js`, while the recap **JSON API** gates on `status = 'archived'`. A published-but-unarchived past event would get SSR meta while its data API 404s. Pre-existing (`sitemap.xml.js` already emits those URLs on `is_published` alone) — filed separately rather than silently tightened here.
- Docs: `CLAUDE.md` gains the append-vs-replace invariant, per the repo rule that a fix updates its documented invariant in the same change.

Closes #783

---
Built by Sonny · Reviewed by Theo · 🤖 [Claude Code](https://claude.ai/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added server-rendered SEO metadata for public pages, including canonical, Open Graph, Twitter, title, and description tags.
  * Added optimized event recap metadata with event-specific URLs, descriptions, and poster images.
  * Added public handling and route coverage for Artists, Venues, About, Contact, Stats, Privacy, Terms, Subscribe, and event recap pages.

* **Bug Fixes**
  * Prevented duplicate or conflicting homepage metadata from appearing on rendered pages.

* **Tests**
  * Added comprehensive coverage for metadata, routing, and fallback behaviour.

### CODE REVIEW

File: functions/utils/staticPageMeta.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      `<meta property="og:url" content="${escapeAttr(url)}" />`,
+      `<meta property="og:title" content="${escapeAttr(ogTitle)}" />`,
+      `<meta property="og:description" content="${escapeAttr(ogDescription)}" />`,
+      `<meta property="og:type" content="website" />`,
+      `<meta property="og:image" content="${escapeAttr(DEFAULT_OG_IMAGE)}" />`,
+      `<meta name="twitter:card" content="summary_large_image" />`,
+      `<meta name="twitter:title" content="${escapeAttr(ogTitle)}" />`,
+      `<meta name="twitter:description" content="${escapeAttr(ogDescription)}" />`,
+      `<meta name="twitter:image" content="${escapeAttr(DEFAULT_OG_IMAGE)}" />`,
+    ];
```
[the remark this answers is not in the document]
Comment (BreakableHoodie) [reply functions/utils/staticPageMeta.js:121]: Fixed in 988c791: inverted meta ownership rather than trying to make Helmet adopt SSR tags. Every SSR-injected route's client `<Helmet>` now declares `<title>` (+ JSON-LD where present) only — canonical/og:*/twitter:*/description are SSR-only across `functions/utils/staticPageMeta.js`, `functions/events/[slug]/recap.js`, and `frontend/src/pages/EventRecapPage.jsx` (plus the other SSR-injected routes swept in the same pass: static pages, `/band/*`, `/venue/*`, `/event/*`).

Rejected `data-rh="true"` on SSR tags as the alternative — it would make correctness depend on Helmet reliably adopting/replacing tags it didn't create, the same behavior CLAUDE.md already documents as unreliable for `document.title` under React 19.

Added a hydrated-DOM regression test (`frontend/src/pages/__tests__/ssrIdentityMeta.test.jsx`) that seeds `document.head` with the tags SSR would inject, mounts each real page component via `createRoot` (matching `main.jsx`), and asserts exactly one canonical/og:url survives across all 10 affected routes. Mutation-checked by hand: reverting any one ownership trim turns its assertion into "found 2" and fails.

Also swept for tags that existed only in a client Helmet with no SSR equivalent (`og:site_name` was missing on 3 routes) so the trim didn't silently drop them instead of de-duplicating — each has its own backend test proving exactly one, not zero.

File: functions/events/[slug]/recap.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    return env.ASSETS.fetch(request);
+  }
+
+  let event;
+  try {
+    event = await env.DB.prepare(
+      `SELECT id, name, slug, date, poster_url
+       FROM events
+       WHERE slug = ? AND (is_published = 1 OR status = 'archived')`,
+    )
```
[the remark this answers is not in the document]
Comment (BreakableHoodie) [reply functions/events/[slug]/recap.js:45]: Confirmed real and left deliberately out of scope for this PR — the recap SSR gate (`is_published = 1 OR status = 'archived'`) is genuinely looser than the recap JSON API's (`status = 'archived'` only), so a published-but-unarchived past event can get recap-specific SSR meta while the client-side fetch 404s.

Not folding it into this PR because it's an independent gating bug (an existing mismatch between two D1 queries), not part of the Helmet-ownership defect this PR targets, and tightening it needs its own test coverage of the `is_published=1`-but-not-archived boundary. Filed as **#791** to track and fix separately.

File: functions/utils/staticPageMeta.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      `<link rel="canonical" href="${escapeAttr(url)}" />`,
+      `<meta name="description" content="${escapeAttr(description)}" />`,
+      `<meta property="og:url" content="${escapeAttr(url)}" />`,
+      `<meta property="og:title" content="${escapeAttr(ogTitle)}" />`,
+      `<meta property="og:description" content="${escapeAttr(ogDescription)}" />`,
+      `<meta property="og:type" content="website" />`,
+      `<meta property="og:image" content="${escapeAttr(DEFAULT_OG_IMAGE)}" />`,
+      `<meta name="twitter:card" content="summary_large_image" />`,
+      `<meta name="twitter:title" content="${escapeAttr(ogTitle)}" />`,
+      `<meta name="twitter:description" content="${escapeAttr(ogDescription)}" />`,
```
[the remark this answers is not in the document]
Comment (BreakableHoodie) [reply functions/utils/staticPageMeta.js:119]: Resolved as a side effect of the ownership fix in 988c791: `SubscribePage.jsx`'s client `<Helmet>` no longer declares `twitter:card` (or any identity/social meta) at all — that block was trimmed to `<title>` only, since SSR (`functions/utils/staticPageMeta.js`'s `STATIC_PAGES["/subscribe"]` entry) is now the single owner. So there's only one `twitter:card` tag on `/subscribe`, and it's SSR's `summary_large_image` (with a real `twitter:image`, unlike the old imageless client value) — the conflicting-values scenario this comment flagged can no longer occur.
